mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIUploadHelper.java

```diff
@@ -146,119 +145,172 @@ public static Response uploadObject(UriInfo info, HttpServletRequest request, In
-                mcrDerivate.getDerivate().setLinkMeta(new MCRMetaLinkID("linkmeta", mcrObjIDObj, null, null));
-                mcrDerivate.getDerivate().setInternals(new MCRMetaIFS("internal", null));
-
-                if (classifications != null && classifications.length() > 0) {
-                    final List<MCRMetaClassification> currentClassifications;
-                    currentClassifications = mcrDerivate.getDerivate().getClassifications();
-                    Stream.of(classifications.split(" "))
-                        .map(MCRCategoryID::fromString)
-                        .filter(dao::exist)
-                        .map(categoryID -> new MCRMetaClassification("classification", 0, null, categoryID))
-                        .forEach(currentClassifications::add);
-                }
-
-                MCRMetadataManager.create(mcrDerivate);
-                MCRMetadataManager.addOrUpdateDerivateToObject(mcrObjIDObj,
-                    MCRMetaEnrichedLinkIDFactory.getInstance().getDerivateLink(mcrDerivate),
-                    mcrDerivate.isImportMode());
-                derID = mcrDerivate.getId();
+                derID = createNewDerivate(mcrObjIDObj, label, classifications);
             }
 
-            response = Response
-                .created(info.getBaseUriBuilder().path("objects/" + mcrObjID + "/derivates/" + derID).build())
-                .type("application/xml; charset=UTF-8")
-                .build();
+            return buildResponse(info, mcrObjIDObj, String.valueOf(derID));
+
         } catch (Exception e) {
-            LOGGER.error("Exeption while uploading derivate", e);
+            LOGGER.error("Exception while uploading derivate", e);
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+        }
+    }
+
+    private static MCRObjectID findDerivateIDIfExists(MCRObject mcrObj, String label, String classifications,
+        boolean overwriteOnExisting) throws MCRRestAPIException {
+        MCRObjectID derID = null;
+        final MCRCategoryDAO dao = MCRCategoryDAOFactory.getInstance();
+
+        if (overwriteOnExisting) {
+            final List<MCRMetaEnrichedLinkID> currentDerivates = mcrObj.getStructure().getDerivates();
+            derID = findDerivateByLabel(currentDerivates, label);
+
+            if (derID == null && classifications != null && !classifications.isEmpty()) {
+                derID = findDerivateByClassifications(currentDerivates, classifications, dao);
+            }
+        }
+
+        return derID;
+    }
+
+    private static MCRObjectID findDerivateByLabel(List<MCRMetaEnrichedLinkID> derivates, String label) {
+        if (label != null && !label.isEmpty()) {
+            for (MCRMetaLinkID derLink : derivates) {
+                if (label.equals(derLink.getXLinkLabel()) || label.equals(derLink.getXLinkTitle())) {
+                    return derLink.getXLinkHrefID();
+                }
+            }
+        }
```

Comment:
Move label condition to parent method. Method name does not match, no derivate is returned.